rcore/overlay_manager.c

```diff
@@ -0,0 +1,218 @@
+    thread->status = AppThreadLoaded;
+    thread->thread_entry = &_overlay_thread;
+    /* start the thread 
+     * We are only using the thread launcher in appmanager for this
+     * not the full supervisory process. It's lightweight
+     */
+    appmanager_execute_app(thread, 0);    
+}
+
+/*
+ * Create a new top level window and all of the contents therein
+ */
+void overlay_window_create(OverlayCreateCallback creation_callback)
+{
+    OverlayMessage om = (OverlayMessage) {
+        .command = OVERLAY_CREATE,
+        .data = (void *)creation_callback
+    };
+    xQueueSendToBack(_overlay_queue, &om, 0);
+}
+
+void overlay_window_draw(void)
+{
+    OverlayMessage om = (OverlayMessage) {
+        .command = OVERLAY_DRAW,
+    };
+    xQueueSendToBack(_overlay_queue, &om, 0);
+}
+
+
+void overlay_window_destroy(OverlayWindow *overlay_window)
+{
+        OverlayMessage om = (OverlayMessage) {
+        .command = OVERLAY_DESTROY,
+        .data = (void *)overlay_window
+    };
+    xQueueSendToBack(_overlay_queue, &om, 0);
+}
+
+list_head *overlay_window_thread_get_head(Window *window)
+{
+    OverlayWindow *ow;
+    Window *w;
+    list_foreach(ow, &_overlay_window_list_head, OverlayWindow, node)
+    {
+        list_foreach(w, &ow->window_list_head, Window, node)
+        {
+            if (w == window)
+            {
+                return &ow->window_list_head;
+            }
+        }
+    }
+    
+    return NULL;
+}
+
+Window *overlay_window_stack_get_top_window(OverlayWindow *overlay_window)
+{
+    list_head *lh = &overlay_window->window_list_head;
+    return list_elem(list_get_head(lh), Window, node);
+}
+
+OverlayWindow *overlay_stack_get_top_overlay_window(void)
+{
+    return list_elem(list_get_head(&_overlay_window_list_head), OverlayWindow, node);
+}
+
+list_head *overlay_window_get_list_head(void)
+{
+    return &_overlay_window_list_head;
+}
+
+uint8_t overlay_window_count(void)
+{
+    uint16_t count = 0;
+    
+    OverlayWindow *w;
+    list_foreach(w, &_overlay_window_list_head, OverlayWindow, node)
+    {
+        count++;
+    }
+    SYS_LOG("window", APP_LOG_LEVEL_ERROR, "OVERLAY COUNT %d.", count);
+    
+    return count;
+}
+
+void overlay_window_stack_push(OverlayWindow *overlay_window, bool animated)
+{
+    list_init_node(&overlay_window->node);
+    list_insert_head(&_overlay_window_list_head, &overlay_window->node);
+    
+    Window* window = overlay_window_stack_get_top_window(overlay_window);
+    assert(window);
+    window->is_render_scheduled = true;
+    window_dirty(true);
+}
+
+void overlay_window_stack_push_window(OverlayWindow *overlay_window, Window *window, bool animated)
+{
+    list_init_node(&window->node);
+    list_insert_head(&overlay_window->window_list_head, &window->node);
+    window_stack_push_configure(window, animated);
+}
+
+
+static void _overlay_window_create(OverlayCreateCallback create_callback)
+{
+    OverlayWindow *overlay_window = app_calloc(1, sizeof(OverlayWindow));
```

Comment:
Which ends up being called in overlay context, right?

Reply:
right